lib/ctblgrp.gi

```diff
@@ -336,6 +309,7 @@ DxRegisterModularChar := function(D,c)
 local d,p;
   # it may happen,that an irreducible character will be registered twice:
   # 2-dim space,1 Orbit,combinatoric split. Avoid this!
+  if IsPlistRep(c) then c:=Vector(c); fi;
```

Comment:
Are the entries of `D.modulars` modified later on, or will they be immutable? In the latter case, you could also try
```suggestion
  if IsPlistRep(c) then c:=ImmutableVector(SOME_RING, c); fi;
```
provided `SOME_RING` can be derived from `D` or can be passed in some other way to this function.

Even if it is mutable, it might be a good idea to pass the basering and/or filter: The problem with `Vector(c)` is that if one passes a list-of-GF(4) elements to it, then sometimes one will get a GF(4) vector yet sometimes a GF(2) vector, which usually is not desirable